inference-engine/src/mkldnn_plugin/ngraph_transformations/rnn_sequences_optimization.cpp

```diff
@@ -113,22 +122,39 @@ MKLDNNPlugin::OptimizeRNNSequenceTransposes::OptimizeRNNSequenceTransposes() {
 
 MKLDNNPlugin::OptimizeLSTMSequenceTransposes::OptimizeLSTMSequenceTransposes() {
     ngraph::matcher_pass_callback callback = [](ngraph::pattern::Matcher &m) {
-        auto lstmSequence = std::dynamic_pointer_cast<ngraph::op::v5::LSTMSequence>(m.get_match_root());
+        std::shared_ptr<ngraph::Node> lstmSequence;
+        // Bidirectional cases are not supported
+        auto lstmSequence5 = std::dynamic_pointer_cast<ngraph::op::v5::LSTMSequence>(m.get_match_root());
```

Comment:
I guess it's better to implement lambda here:
```
auto get_seq = [] (const std::shared_ptr<Node>& node) -> bool { 
if (auto lstm_5 = std::dynamic_pointer_cast<...>(node)) {
 return lstm_5->get_direction() == ngraph::op::RecurrentSequenceDirection::BIDIRECTIONAL;
} else if  (auto lstm_0 = std::dynamic_pointer_cast<...>(node)) {
 return lstm_0->get_direction() == ngraph::op::RecurrentSequenceDirection::BIDIRECTIONAL;
} else {
return false;
}

```

Reply:
even not lambda, just if - else if - else

Reply:
Done, thanks!